### ISSUE TITLE
[bitnami/spring-cloud-dataflow] Multiple external kafka nodes

### ISSUE THREAD

**Issue by ljohnston** [bitnami/charts#7639]
Chart:  spring-cloud-dataflow v4.1.0

Configuring the chart to use multiple kafka brokers via something like the following in `values.yaml` does not work correctly when deploying a stream application.
```
externalKafka:
  enabled: true
  brokers: kafka-node-1:9092,kafka-node-2:9092,kafka-node-3:9092
  zkNodes: zk-node-1:2181,zk-node-2:2181,zk-node-3:2181
```
With the above configuration in place trying to deploy a stream application fails with the error message:
```
Invalid environment variable declared: kafka-node-2:9092
```
**Steps to reproduce the behavior**

1. Configure and deploy the helm chart with multiple external kafka brokers configured as shown above.
2. Try and deploy a stream application (such as `time | log`)
3. The deployment will fail with the `Invalid environment variable ...` error shown above.

**Expected behavior**
The stream application should deploy successfully.

**Version of Helm and Kubernetes**

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.6.0", GitCommit:"7f2df6467771a75f5646b7f12afb408590ed1755", GitTreeState:"clean", GoVersion:"go1.16.3"}
```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.17", GitCommit:"f3abc15296f3a3f54e4ee42e830c61047b13895f", GitTreeState:"clean", BuildDate:"2021-01-13T13:21:12Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17+", GitVersion:"v1.17.17-eks-087e67", GitCommit:"087e67e479962798594218dc6d99923f410c145e", GitTreeState:"clean", BuildDate:"2021-07-31T01:39:55Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**
- I tried several variations of single/double quoting the `externalKafak.brokers` value but was unable to get anything to work.
- Adding the application property `spring.cloud.stream.kafka.binder.brokers` in the dataflow server UI at deployment time, specifying multiple comma-delimited kafka nodes, works as expected.
- I would expect the same issue occurs for the `externalKafka.zkNodes` chart value, but have not confirmed that.

**Comment by juan131** [comment]
Hi @ljohnston 

Thanks so much for reporting the issue!!!

Could you try using the values.yam below and share with us the results?

```yaml
externalKafka:
  enabled: true
  brokers: '"kafka-node-1:9092,kafka-node-2:9092,kafka-node-3:9092"'
  zkNodes: '"zk-node-1:2181,zk-node-2:2181,zk-node-3:2181"'
```

**Comment by juan131** [comment]
Hi @ljohnston

I used this **values.yaml**:

```yaml
rabbitmq:
  enabled: false
kafka:
  enabled: false
externalKafka:
  enabled: true
  brokers: '"kafka-node-1:9092,kafka-node-2:9092,kafka-node-3:9092"'
  zkNodes: zk-node-1:2181,zk-node-2:2181,zk-node-3:2181
```

As you can see I'm using the `'"` workaround for _brokers_ while keeping the value for _zkNodes_ without any "trick". Then, I run the command below:

```console
$ helm template spring-cloud-dataflow bitnami/spring-cloud-dataflow -f values.yaml -s templates/skipper/configmap.yaml
(...)
data:
  application.yaml: |-
(...)
                    environmentVariables: 'SPRING_CLOUD_STREAM_KAFKA_BINDER_BROKERS="kafka-node-1:9092,kafka-node-2:9092,kafka-node-3:9092",SPRING_CLOUD_STREAM_KAFKA_BINDER_ZK_NODES=zk-node-1:2181,zk-node-2:2181,zk-node-3:2181'
(...)
```

As you can see, the env. variable `SPRING_CLOUD_STREAM_KAFKA_BINDER_BROKERS` makes a proper use of the quotes when assigning the value, while the `SPRING_CLOUD_STREAM_KAFKA_BINDER_ZK_NODES` does not.

**Comment by ljohnston** [comment]
Hi @juan131 

Yes, I get the same result. And spring dataflow gets deployed without issue. If, however, I then go into the UI and try and deploy a simple stream app (e.g. `time | log`), the deployment will fail with the error message I mentioned.
```
Invalid environment variable declared: kafka-node-2:9092
```
Thanks,
Lance

**Comment by juan131** [comment]
Hi @ljohnston 

I think we'll need extra quotes to avoid Skipper to recognize the `SPRING_CLOUD_STREAM_KAFKA_BINDER_BROKERS` value as different env. variables. Using the **values.yaml** I shared before, I access the Skipper pod and took a look to the **application.yml** using `yq`:

```console
$ yq eval .spring.cloud.skipper.server.platform.kubernetes.accounts.default.environmentVariables /opt/bitnami/spring-cloud-skipper/conf/application.yml
SPRING_CLOUD_STREAM_KAFKA_BINDER_BROKERS="kafka-node-1:9092,kafka-node-2:9092,kafka-node-3:9092",SPRING_CLOUD_STREAM_KAFKA_BINDER_ZK_NODES=zk-node-1:2181,zk-node-2:2181,zk-node-3:2181
```

We might need to escape the quotes...  Could you please provide (step by step) how you're deploying your stream app through the UI?

**Comment by ljohnston** [comment]
Hi @juan131 

Sure... after deploying spring dataflow via the chart, I do the following in the UI:
* Go to "Application" on the left hand nav menu.
* Click "Add application(s)".
* Under "Import application starters from dataflow.spring.io" click "Stream application starters for Kafka/Docker".
* Click "Import application(s)".
* Go to "Streams" on the left hand nav menu.
* Click "Create stream(s)".
* Enter "time | log" for the stream definition.
* Click "Create stream(s)".
* Enter "ticktock" for the name.
* Click "Create the stream(s)".
* Click the "ticktock" stream.
* Click "Deploy stream".
* On the deploy stream page, select "default (kuberentes)" for the Platform in the "Global" column.
* Click the "Deploy stream" button at the bottom of the page.
* An exception dialog pops out from the right hand side of the page. The bottom of that exception will include the "Invalid environment variable declared..." message.

Let me know if you need anything else.

Appreciate your help with this.

Thanks,
Lance

**Comment by juan131** [comment]
Hi @ljohnston 

Thanks so much for sharing the exact steps! I could reproduce the issue following them.

I've been trying with different combination of values without luck. These are just a few examples:

```yaml
...
  brokers: "'\"kafka-node-1:9092,kafka-node-2:9092,kafka-node-3:9092\"'"
  zkNodes: "'\"zk-node-1:2181,zk-node-2:2181,zk-node-3:2181\"'"
```

```yaml
...
  brokers: "'\\\"kafka-node-1:9092,kafka-node-2:9092,kafka-node-3:9092\\\"'"
  zkNodes: "'\\\"zk-node-1:2181,zk-node-2:2181,zk-node-3:2181\\\"'"
```

```yaml
...
  brokers: '""kafka-node-1:9092,kafka-node-2:9092,kafka-node-3:9092""'
  zkNodes: '""zk-node-1:2181,zk-node-2:2181,zk-node-3:2181""'
```

I will create an internal task to investigate this issue properly.

**Comment by fmulero** [comment]
Hi @ljohnston 

We've just released a new version of the chart (4.2.3) with a solution for this problem. In that version you can use your original `values.yaml` without adding any extra quotes

Please give it a try when you have a chance and provide us feedback.
